src/routes/synced_services_config.ts

```diff
@@ -154,4 +168,230 @@ router.get('/toggl_track_workspaces', authCommons.checkJwt, async (req, res) =>
   }
 });
 
+/*
+  check if webhook exist
+    if does, return 304
+  else create new webhook for the connection
+*/
+router.post('/toggl_track_create_webhook', authCommons.checkJwt, async (req, res) => {
+  const togglTrackApiKey: string | undefined = req.body['api_key']?.toString();
+  const workspaceId: string | undefined = req.body['workspaceId']?.toString()
+  const connectionId: string | undefined = req.body['connectionId']?.toString()
+  const ApiURL: string | undefined = req.body['backendURL']?.toString()
+
+  if (!togglTrackApiKey || !workspaceId || !connectionId || !ApiURL) {
+    return res.sendStatus(400);
+  }
+
+  const callbackUrl: string = `${ApiURL}webhooks/toggl_track/${connectionId}`
+  //check if exists
+  const webhooksResponse = await getExistingTogglTrackWebhooks(togglTrackApiKey, workspaceId)
+  if (!webhooksResponse || typeof webhooksResponse === 'number') {
+    let statusCode = 503;
+    if (webhooksResponse && webhooksResponse !== 401) {
+      statusCode = webhooksResponse;
+    } else if (webhooksResponse && webhooksResponse === 401) {
+      // do not send 401, it would lead to user logout on the client side due to error intercepting
```

Comment:
Tohle prosím o vysvětlení. Pokud něco takového client dělá tak je to blbě. API by podle mě mělo vrátit 401.

Reply:
mělo by se řešit na FE